### PR TITLE
Haiii :3

### PR DESCRIPTION
**Pull request by LucioGoodTimes**
It adds a "search autocomplete" feature

like when you're typing in a search box in a booru and it suggests tags for you

documented all my incredible discoveries in this comment because i like talking to myself:
https://github.com/LucioMacarine/BooruSharp/commit/3efd5d8d2fe7fad6bfbb787a89585e362475bbbb#commitcomment-87519255

### CODE REVIEW

**Comment by Xwilarg** [comment]
Thanks for your PR, I'll have a closer look at it tonight (in like 8 hours)

File: BooruSharp/Booru/ABooru.cs [hunk 1]
```diff
@@ -198,6 +205,23 @@ protected ABooru(string domain, UrlFormat format, BooruOptions options)
...
+                _autocompleteUrl = format == UrlFormat.IndexPhp
+                    ? new Uri(BaseUrl + "autocomplete.php")
+                    : CreateQueryString(format, "autocomplete");
+            switch (_format)
+            {
+                case UrlFormat.IndexPhp:
+                    _autocompleteUrl = new Uri(BaseUrl + "autocomplete.php");
+                    break;
+                case UrlFormat.Danbooru:
+                    _autocompleteUrl = new Uri(BaseUrl + "tags/" + "autocomplete.json");
```
Comment (Xwilarg) [inline BooruSharp/Booru/ABooru.cs]:
```suggestion
                    _autocompleteUrl = new Uri(BaseUrl + "tags/autocomplete.json");
```

File: BooruSharp/Booru/Template/E621.cs [hunk 2]
```diff
@@ -136,5 +136,36 @@ private protected override Search.Tag.SearchResult GetTagSearchResult(object jso
...
+        {
+            var elem = (JArray)json;
+            var autoCompleteResults = new List<Search.Autocomplete.SearchResult>();
+            foreach (var item in elem.Children())
+            {
+                int id = item["id"].Value<int>();
+                string name = item["name"].Value<string>();
+                int count = item["post_count"].Value<int>();
+                var type = GetTagType(item["category"].Value<int>());
+                string antecedent_name = item["antecedent_name"].Value<string>();
```
Comment (Xwilarg) [inline BooruSharp/Booru/Template/E621.cs]: Please use camelCase for variable names

**Review by Xwilarg — CHANGES_REQUESTED** [review]
Just had a quick look at the code, will try it next, just marked 2 small things for changes

**Comment by Xwilarg** [comment]
Oh could I also ask you to update the table ( https://github.com/Xwilarg/BooruSharp#features-availability ) in the README?

**Comment by LucioGoodTimes** [comment]
Done.

**Comment by LucioGoodTimes** [comment]
Also added an example in the README if you don't mind

**Comment by Xwilarg** [comment]
Hey sorry late answer, I'll have a last try tomorrow then merge this (+ nuget packages and stuff)

**Review by Xwilarg — APPROVED** [review]

**Pull request merged**